doc/architecture/decisions/0002-object-identifiers.md

```diff
@@ -0,0 +1,57 @@
+# 0002. Object identifiers
+
+Date: 2019-06-11
+
+## Status
+
+Agreed
+
+## Context
+
+To uniquely identify an object in Ensembl we need three pieces of information:
+
+-   The **dataset** to which the object belongs, e.g. homo_sapiens_GCA_000001405_14 [^1]
+-   The **object type**, e.g. gene, transcript, region  
+-   The **object ID**, e.g. ENSG00000139618, AT3G52430, 13:32315474-32400266 [^2]  
```

Comment:
Ah we're also referring to region as an ID. Okay then this makes more sense to use the `assembly_id`. We do need to watch out here because semantically the three classes of data you've mentioned here are not interchangeable.